File: Modules/SIE.ZYF/SIE.Web.ZYF/PurchaseOrders/Commands/AddPurOrderCommand.cs
using SIE.Web.Command;
using SIE.ZYF.PurchaseOrders;

namespace SIE.Web.ZYF.PurchaseOrders.Commands
{
    public class AddPurOrderCommand : ViewCommand
    {
        protected override object Excute(ViewArgs args, string scope)
        {
            var productManage = args.Data.ToJsonObject<PurchaseOrder>();
            productManage.OrderNumber = RT.Service.Resolve<PuOrderController>().GetOrderNumber();
            return productManage.OrderNumber;
        }
    }

}

File: Modules/SIE.ZYF/SIE.Web.ZYF/PurchaseOrders/Commands/AddPurOrderCommand.js
SIE.defineCommand("SIE.Web.ZYF.PurchaseOrders.Commands.AddPurOrderCommand", {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加", group: "edit", iconCls: "icon-AddEntity icon-green" },
    showView: function (entity) {
        var model = entity.data;
        var me = this;
        this.view.execute({
            data: model,
            success: function (res) {
                var number = res.Result;
                CRT.Workbench.addPage({
                    entityType: me.view.model,
                    recordId: entity.getId(),
                    viewGroup: "DetailsView",
                    title: me.getEditViewTitle(entity),
                    params: {
                        Number: number
                    },
                    isDetail: true,
                })
            }
        }, me.view);
    }
});